### PR TITLE
samples: fix conformance issues in nrf sample.yaml files

### PR DESCRIPTION
**Pull request by frkv**
-Fixing compliance in sample.yamle for applications and samples for
 bluetooth, CAF, crypto, hw_id, matter, remote_shell location,
 openthread, peripheral, zigbee and nfc.
-Updated sample.yaml files to conform to the lint rules on line-length -Utilizing yaml "folding" with ">" for platform_allow and  extra_args split into multiple lines without spurious newlines added.
-Utilizing yaml "chomping" with "|" for descriptions that needed to be split up in multiple lines.
-Fix some indentation issues in psa_tls and crypto samples

Note: The only sample not touched is nrf/samples/nrf9160/modem_shell as this has more complex quoted inputs for overlays that are too long

Signed-off-by: Frank Audun Kvamtrø <frank.kvamtro@nordicsemi.no>

### CODE REVIEW

**Review by tokangas — APPROVED** [review]
Looks fine for the nrf9160/location sample.

**Review by maxd-nordic — APPROVED** [review]

**Review by tomchy — APPROVED** [review]

**Review by trantanen — APPROVED** [review]
Location sample is fine.

**Review by sebastiandraus — APPROVED** [review]

File: samples/crypto/persistent_key_usage/sample.yaml [hunk 1]
```diff
@@ -1,20 +1,23 @@
 sample:
-    description: This app provides an example of using peristent keys with the
-        PSA APIs. A random AES persistent key is generated and imported
-        to the PSA keystore.
-    name: Persistent key example
+  description:
```
Comment (MarekPieta) [inline samples/crypto/persistent_key_usage/sample.yaml:2]: Is the `|` missing here? Or it's intentional ?

**Review by MarekPieta — APPROVED** [review]
Looks good. One small comment

File: samples/crypto/persistent_key_usage/sample.yaml (hunk 1 above)
Comment (frkv) [reply samples/crypto/persistent_key_usage/sample.yaml:2]: Oh sorry. This one seems to be missing

Comment (frkv) [reply samples/crypto/persistent_key_usage/sample.yaml]: Fixed. Rerunning CI

**Pull request merged**